main.go

```diff
@@ -145,6 +145,10 @@ func main() {
 		config.WithLabelFilters(labelsFilter),
 	)
 
+	if err := cfg.AutoDetect(); err != nil {
```

Comment:
The idea of auto-detect in the OTEL operator is that APIs can be dynamically installed in the cluster (e.g. after the operator was installed). 

Could you please elaborate why the initialization should block until the first auto detect cycle is executed?